### PR TITLE
release-tool: automate indexer constants update

### PR DESCRIPTION
**Pull request by coury-clark**
Closes https://github.com/sourcegraph/sourcegraph/issues/48862

Automates the indexer constants in the release branch and main branch during bake step.

## Test plan

I tested locally by pointing the release tool at my local sourcegraph repo and running the command.

## App preview:

- [Web](https://sg-web-cclark-automate-indexer-constants.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: dev/release/src/main.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ async function main(): Promise<void> {
 
     const step = args[0] as StepID
     const stepArguments = args.slice(1)
-    ensureMainBranchUpToDate()
+    // ensureMainBranchUpToDate()
```
Comment (coury-clark) [inline dev/release/src/main.ts:19]:
```suggestion
    // ensureMainBranchUpToDate()
```

File: dev/release/src/release.ts [hunk 2]
```diff
@@ -1003,7 +1003,7 @@ ${patchRequestIssues.map(issue => `* #${issue.number}`).join('\n')}`
 
             const releaseBranch = release.branch
             const version = release.version.version
-            ensureReleaseBranchUpToDate(releaseBranch)
+            // ensureReleaseBranchUpToDate(releaseBranch)
```
Comment (coury-clark) [inline dev/release/src/release.ts:1006]:
```suggestion
            ensureReleaseBranchUpToDate(releaseBranch)
```

File: dev/release/src/util.ts [hunk 3]
```diff
@@ -323,19 +324,25 @@ export async function validateNoReleaseBlockers(octokit: Octokit): Promise<void>
...
-    if (!repoPath) {
-        const client = await getAuthenticatedGitHubClient()
-        const { workdir } = await cloneRepo(client, 'sourcegraph', 'src-cli', {
-            revision: 'main',
-            revisionMustExist: true,
-        })
-        repoPath = workdir
+export async function nextSrcCliVersionInputWithAutodetect(config: ReleaseConfig, repoPath?: string): Promise<SemVer> {
+    let next: SemVer
+    if (!config.in_progress?.srcCliVersion) {
```
Comment (coury-clark) [inline dev/release/src/util.ts:329]: This is somewhat unrelated, but it was bothering me while testing that the src-cli detection wouldn't use the version in the config.

**Review by eseliger — APPROVED** [review]
LGTM, I've also tagged Eric for another pair of eyes, to make sure this is sane. Thanks so much for doing all these little QoL things for us :)

**Review by efritz — APPROVED** [review]

**Pull request merged**